include/kframework/lemmas/lemmas.k

```diff
@@ -80,12 +82,19 @@ module LEMMAS [symbolic]
   // Map Reasoning
   // ########################
 
-    rule #lookup ( _M:Map [ K1 <- V1 ] , K2 ) => #lookup ( K1 |-> V1 , K1 )  requires K1 ==Int  K2 [simplification]
-    rule #lookup (  M:Map [ K1 <- _  ] , K2 ) => #lookup ( M         , K2 )  requires K1 =/=Int K2 [simplification]
+    rule keys(.Map)             => .Set                    [simplification]
+    rule keys(M [ K <- _V ])    => keys(M) |Set SetItem(K) [simplification]
+    rule keys(M [ K <- undef ]) => keys(M) -Set SetItem(K) [simplification]
 
     rule 0 <=Int #lookup( _M:Map , _ )             => true [simplification, smt-lemma]
     rule         #lookup( _M:Map , _ ) <Int pow256 => true [simplification, smt-lemma]
 
+    rule #lookup(M:Map, X:Int) => 0 requires notBool (X:Int in keys(M):Set) [simplification]
```

Comment:
What's the difference between this lemma and [this rule](https://github.com/runtimeverification/evm-semantics/blob/ba06470f78f9b37ff9058b62a87c749a5c7aee13/include/kframework/evm-types.md?plain=1#L414) in the definition of `#lookup`?

Reply:
The difference is that `in_keys` is a `K` built-in. I wasn't able to use it well in Lukso for some reason and so I created `keys` instead.